lib/utils/TreePublisher.ts

```diff
@@ -0,0 +1,13 @@
+import {BlueshellState} from '../nodes/BlueshellState';
+
+export interface TreePublisher<B extends BlueshellState, V> {
+	maybePublishTree(state: B, event: V, topLevel: boolean): void;
```

Comment:
this is mostly a stylistic comment, but why is this `maybe`? that's an implementation detail of the implementing function, as far as the caller is concerned its getting published. Also, its obviously publishing a tree because its a TreePublisher, its more that its publishing changes.

I'd call this function `TreePublisher.publishResult()` or `logResult()`

Reply:
Change adopted.